realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/MongoClientTest.kt

```diff
@@ -876,6 +884,363 @@ class MongoClientTest {
         }
     }
 
+    fun assertDocumentEquals(expected: Document, actual: Document) {
+        assertEquals(expected.keys.size, actual.keys.size - 1)
+
+        for (key in expected.keys) {
+            assertTrue(actual.keys.contains(key))
+            assertEquals(expected[key], actual[key])
+        }
+    }
+
+    @Test
+    fun watchStreamSynchronous() {
+        looperThread.runBlocking {
+            with(getCollectionInternal()) {
+
+                val insertedDocument = Document("watch", "1")
+                        .apply {
+                            this["num"] = 1
+                        }
+
+                val updatedDocument = Document("watch", "1")
+                        .apply {
+                            this["num"] = 2
+                        }
+
+
+                val watcher = this.watch()
+
+                thread {
+                    watcher.next.let { changeEvent ->
+                        assertEquals(OperationType.INSERT, changeEvent.operationType)
+                        assertDocumentEquals(insertedDocument, changeEvent.fullDocument!!)
+                    }
+
+                    watcher.next.let { changeEvent ->
+                        assertEquals(OperationType.REPLACE, changeEvent.operationType)
+                        assertDocumentEquals(updatedDocument, changeEvent.fullDocument!!)
+                    }
+
+                    watcher.next.let { changeEvent ->
+                        assertEquals(OperationType.DELETE, changeEvent.operationType)
+                        assertNull(changeEvent.fullDocument)
+                    }
+
+                    looperThread.testComplete()
+                }
+
+                while (!watcher.isOpen) {
```

Comment:
Nice that it works. I would still write a short explanation as to why you added this. Seeing a loop with an empty body could look a bit weird or intimidating to some.